### PR TITLE
NAS-122137 / 23.10 / Fix reporting dropdown menu

### PR DESCRIPTION
**Pull request by denysbutenko**
The issue is missing menu items in the dropdown menu, such as UPS and Target, when reports are available.

For testing, find a system with UPS or mock it in the `reports.service.ts`.

```
this.ws.call('reporting.graphs').subscribe((reportingGraphs) => {
      reportingGraphs.push({
        name: 'ups',
        title: 'UPS',
        vertical_label: 'Requests',
        identifiers: [
          'demand_data',
          'demand_metadata',
          'prefetch_data',
          'prefetch_metadata',
        ],
        stacked: true,
        stacked_show_total: true,
      });
      reportingGraphs.push({
        name: 'ctl',
        title: 'Target',
        vertical_label: 'Requests',
        identifiers: [
          'demand_data',
          'demand_metadata',
          'prefetch_data',
          'prefetch_metadata',
        ],
        stacked: true,
        stacked_show_total: true,
      });
      this.hasUps$.next(reportingGraphs.some((graph) => graph.name === ReportingGraphName.Ups));
      this.hasTarget$.next(reportingGraphs.some((graph) => graph.name === ReportingGraphName.Target));
      this.reportingGraphs$.next(reportingGraphs);
    });
```

### CODE REVIEW

File: src/app/pages/reports-dashboard/reports.service.ts [hunk 1]
```diff
@@ -145,22 +145,23 @@ export class ReportsService implements OnDestroy {
...
-        }
-
-        return true;
-      })
-      .map(([value, label]) => {
-        return { value, label } as ReportTab;
-      });
+  getReportTabs(): Observable<ReportTab[]> {
+    return of(Array.from(reportTypeLabels)).pipe(
+      withLatestFrom(this.hasUps$, this.hasTarget$, this.hasDiskTemperature$),
```
Comment (undsoft) [inline src/app/pages/reports-dashboard/reports.service.ts:150]: What was the advantage of making `getReportTabs` return Observable? It's still only going to emit once. Looks like you could do:
```
combineLatest([
       this.reportsService.getReportGraphs(),
       of(this.reportsService.getReportTabs()),
])
```
above.

**Review by undsoft — APPROVED** [review]
Presumably it works (when testing via mocks).

File: src/app/pages/reports-dashboard/reports.service.ts (hunk 1 above)
Comment (denysbutenko) [reply src/app/pages/reports-dashboard/reports.service.ts]: It seems overkill having tabs as an observable. We could rely on the `getReportGraphs` only.

**Review by undsoft — APPROVED** [review]
Looks good.

**Pull request merged**